apps/emqx/src/emqx_persistent_session_ds_state.erl

```diff
@@ -53,7 +60,8 @@
 
 -type subscriptions() :: emqx_topic_gbt:t(_SubId, emqx_persistent_session_ds:subscription()).
 
--opaque session_iterator() :: emqx_persistent_session_ds:id() | '$end_of_table'.
+-opaque session_iterator_inner() :: emqx_persistent_session_ds:id().
+-type session_iterator() :: session_iterator_inner() | '$end_of_table'.
```

Comment:
At first glance it honestly looks essentially equivalent to `-type session_iterator() :: emqx_persistent_session_ds:id() | '$end_of_table'.`.